nautobot/dcim/factory.py

```diff
@@ -0,0 +1,146 @@
+    parent = factory.Maybe("has_parent", random_instance(Region), None)
+    name = factory.Maybe("has_parent", UniqueFaker("city"), UniqueFaker("country"))
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+
+class SiteFactory(DjangoModelFactory):
+    class Meta:
+        model = Site
+        exclude = ("has_region", "has_tenant")
+
+    name = UniqueFaker("street_address")
+    status = random_instance(lambda: Status.objects.get_for_model(Site), allow_null=False)
+
+    has_region = factory.Faker("pybool")
+    region = factory.Maybe("has_region", random_instance(Region), None)
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+
+    time_zone = pytz.timezone("US/Eastern")
+    physical_address = factory.Faker("address")
+    shipping_address = factory.Faker("address")
+    latitude = factory.Faker("latitude")
+    longitude = factory.Faker("longitude")
+    contact_name = factory.Faker("name")
+    contact_phone = factory.Sequence(lambda n: f"1091-65912-{n:04d}")
+    contact_email = factory.Faker("company_email")
+
+
+class LocationTypeFactory(DjangoModelFactory):
+    class Meta:
+        model = LocationType
+        exclude = ("has_description",)
+
+    name = factory.Sequence(
+        lambda n: "Root"
+        if n == 0
+        else (
+            "Building"
+            if n == 1
+            else ("Floor" if n == 2 else ("Elevator" if n == 3 else ("Room" if n == 4 else "Aisle")))
+        )
+    )
+    parent = factory.LazyAttribute(
+        lambda l: LocationType.objects.get(name="Building")
+        if l.name in ["Floor", "Elevator"]
+        else (
+            LocationType.objects.get(name="Floor")
+            if l.name == "Room"
+            else (LocationType.objects.get(name="Room") if l.name == "Aisle" else None)
+        )
+    )
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+    @factory.post_generation
+    def content_types(self, create, extract, **kwargs):
+        """Assign some contenttypes to a location after generation"""
+        if self.name in ["Building", "Floor"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Prefix),
+                    ContentType.objects.get_for_model(VLANGroup),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+        elif self.name in ["Room", "Elevator"]:
+            self.content_types.set([ContentType.objects.get_for_model(VLAN)])
+        elif self.name in ["Aisle"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Device),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
```

Comment:
Let's set the content_types for `Root` to all, i.e. `self.content_types.set(ContentType.objects.filter(FeatureQuery("locations").get_query()))`

Also let's make sure that we have some LocationType capacity for all supported content_types - CircuitTermination, PowerPanel, RackGroup, Rack, Cluster, etc.